src/CppUTestExt/MockSupport.cpp

```diff
@@ -449,7 +449,7 @@ MockSupport* MockSupport::getMockSupportScope(const SimpleString& name)
 
     if (hasData(mockingSupportName)) {
         STRCMP_EQUAL("MockSupport", getData(mockingSupportName).getType().asCharString());
-        return (MockSupport*) getData(mockingSupportName).getObjectPointer();
+        return static_cast<MockSupport*>( const_cast<void*>( getData(mockingSupportName).getObjectPointer() ) );
```

Comment:
I don't think we are using 'new' style casts in the CppUtest codebase. Could you check this?

Reply:
I've put the static cast with the "new" style just for homogeneity with the const cast. I think that const casts can only be performed with the "new" style, but I'll check anyway if clang accepts as const cast a cast that maintains the type but removes only the const expression.

Reply:
Confirmed, doing a 2-step conversion is not accepted as a const cast by clang:
![image](https://user-images.githubusercontent.com/6639296/36482704-703c2a14-1714-11e8-9664-33138dec36c9.png)

Maybe it's the moment to being to introduce "new" style casts... :sweat_smile:

Reply:
As the necessity to perform a const cast is in fact due to the lack of differentiation between const and non-const objects in MockSupport auxiliary data model, the const cast could be totally unnecessary if PR #1141 is merged. In this case I could revert the changes due to const casts, as Clang should not moan any more.

Reply:
Merged. Can we now remove the new style casts again?